compiler/passes/buildDefaultFunctions.cpp

```diff
@@ -386,8 +387,6 @@ static void build_chpl_entry_points() {
   if (fLibraryCompile) {
     if (chpl_user_main)
       INT_FATAL(chpl_user_main, "'main' found when compiling a library");
-    if (mainModules.n != 1)
```

Comment:
I think I had the notion that a library would create a special initialization entry point, in which case it would be nice to be able to detect the "root" module automatically.  But the future implementor of that feature can insert consistency tests as need.  So, there is nothing lost (as far as I can see) in removing this check.